### PR TITLE
docs: Clarify staterror modifier specification

### PR DESCRIPTION
**Pull request by Moelf**
clarify and fix typo for `Staterror` modifier.

# Description
The `\delta` parameter is the absolute error in each bin due to MC Stat at +/- 1 sigma. Thus, the `\sigma` of the Gaussian constraint term should have `\delta / nominal` value not the other way around.

And by play around with the `logpdf()` output, I'm convinced that the input in JSON is `\delta` rather than `\delta^2` (https://scikit-hep.org/pyhf/intro.html#id25)

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Correct the description of the calculation of the scale of the constraint
to indicate it is calculated from the relative statistical uncertainty from the
channel's event rate.
* Add additional channel subscripts to make the notation be more explicit that
staterror is per-channel.
* Remove typo of additional 'are'.
* Add Jerry Ling to contributor list.
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
Thanks for the PR @Moelf. We'll try to get to it before the end of the weekend.

As you didn't make a GitHub Issue for this can you give a full summary in the PR body and then also finish off the rest of the [CONTRIBUTING guidelines](https://github.com/scikit-hep/pyhf/blob/master/CONTRIBUTING.md#pull-request-procedure) (hint: adding yourself to contributors. :wink:)?

**Comment by Moelf** [comment]
I think the `\delta^2` in the Table can be understood as:
1. we take the quadrature sum of `\delta_b` across all the samples with the same MC error
2. we pass the `sqrt(sum)` into the `staterror` constructor.

So I think the table is not wrong.

File: docs/likelihood.rst
Comment (kratsg) [inline docs/likelihood.rst:202]: Well, one can _change_ the constraint for the staterror typically... so I'm slightly concerned about specifically mentioning the distribution here.

File: docs/likelihood.rst
Comment (kratsg) [inline docs/likelihood.rst:203]: same comment here

File: docs/likelihood.rst
Comment (kratsg) [inline docs/likelihood.rst:205]: this deviates from the notation in the table...

File: docs/likelihood.rst
Comment (Moelf) [reply docs/likelihood.rst:202]: what's a possible alternative? I can add `e.g.` maybe?

File: docs/likelihood.rst
Comment (Moelf) [reply docs/likelihood.rst]: true, fixed

File: docs/likelihood.rst
Comment (kratsg) [reply docs/likelihood.rst]: Well, I think we mention Gaussian already originally so maybe... leave this for now.

File: docs/likelihood.rst
Comment (kratsg) [inline docs/likelihood.rst:205]: this is missing the channel, as this is done _per_ channel separately.

**Comment by matthewfeickert** [comment]
Note to myself for when I review this tomorrow: Relevant RTD build section is here: https://pyhf--1856.org.readthedocs.build/en/1856/likelihood.html#mc-statistical-uncertainty-staterror

File: docs/likelihood.rst
Comment (matthewfeickert) [inline docs/likelihood.rst:202]: I will unhelpfully note before I sign off tonight that there is a Sphinx render issue here with `:math:`

sphinx-issue

I don't mind fixing it tomorrow though myself while reviewing. So don't take this as criticism or a demand &mdash; just a note.

**Review by matthewfeickert — CHANGES_REQUESTED** [review]

File: docs/likelihood.rst
Comment (matthewfeickert) [inline docs/likelihood.rst:200]: @kratsg @lukasheinrich I need to refresh myself on this, but if we clarify here that staterror is a per-channel modifier, then why don't we need to additionally denote this in the [Modifiers and Constraints table](https://scikit-hep.org/pyhf/intro.html#tab-modifiers-and-constraints)?

modifiers

Did we have a good reason for this?

File: docs/likelihood.rst
Comment (matthewfeickert) [inline docs/likelihood.rst:210]:
```suggestion
As not all samples within a channel are estimated from MC simulations, only
```

I think we had a typo here in the original docs as well. oops.

**Review by matthewfeickert — CHANGES_REQUESTED** [review]

**Review by matthewfeickert — APPROVED** [review]
@lukasheinrich @kratsg while I think we should maybe discuss more the notational consistency across the docs I don't think that this needs to hold up Jerry's PR. So if we get approvals from both of you then I this LGTM for now.

File: docs/likelihood.rst
Comment (Moelf) [reply docs/likelihood.rst:200]: I think in the end per-bin implies per-channel, since different channels may not have the same binning.

Comment (matthewfeickert) [reply docs/likelihood.rst:200]: > I think in the end per-bin implies per-channel, since different channels may not have the same binning.

Yeah, this is somewhat my point. Though while per-bin notation should imply per-channel, one could also say that being overly explicit probably wouldn't hurt or new users and I don't think would confuse people experienced with the spec.

I could go either way here, so would be good to get @lukasheinrich to weight in given that he is the only one who hasn't yet.

Comment (kratsg) [reply docs/likelihood.rst:200]: > Yeah, this is somewhat my point. Though while per-bin notation should imply per-channel, one could also say that being overly explicit probably wouldn't hurt or new users and I don't think would confuse people experienced with the spec.

This probably depends on how we define the indexing right? Arguably, one could say that `(c,b) = (2, 0)` is equivalent to the flattened coordinate `(b,) = (10,)`. So the notation of `sigma_{cb}` should be equivalent to `sigma_{b}` but the `b` have different interpretations.

**Comment by matthewfeickert** [comment]
@kratsg @lukasheinrich if this can get reviewed today by one (ideally both) of you then I think we can merge this.

**Review by kratsg — APPROVED** [review]

**Comment by matthewfeickert** [comment]
Thanks for your PR @Moelf!

**Pull request merged**
